### PR TITLE
Ruby: Update gemspec and add useful metadata files

### PR DESCRIPTION
**Pull request by landrito**
Based on hand-edits made to [google-cloud-monitoring](https://github.com/GoogleCloudPlatform/google-cloud-ruby/tree/master/google-cloud-monitoring)

### CODE REVIEW

**Comment by landrito** [comment]
cc @lukesneeringer this should help resolve any discrepancies you found with autogenerated metadata files.

File: src/main/resources/com/google/api/codegen/ruby/gemspec.snip [hunk 1]
```diff
@@ -2,36 +2,33 @@
...
-      s.add_dependency "{@packageDep.name}", "~> {@packageDep.versionBound.lower}"
+      gem.add_dependency "{@packageDep.name}", "~> {@packageDep.versionBound.lower}"
     @end
 
-    s.add_development_dependency "bundler", "~> 1.9"
-    s.add_development_dependency "rake", "~> 10.4"
-    s.add_development_dependency "rubocop", "~> 0.32"
-    s.add_development_dependency "simplecov", "~> 0.9"
-    s.add_development_dependency "minitest", "~> 5.10"
+    gem.add_development_dependency "minitest", "~> 5.10"
```
Comment (geigerj) [inline src/main/resources/com/google/api/codegen/ruby/gemspec.snip]: Should these versions be configurable through the yamls in `googleapis`? No need to address this in this PR, but if so, please file an issue.

File: src/main/resources/com/google/api/codegen/ruby/gitignore.snip [hunk 2]
```diff
@@ -0,0 +1,11 @@
+@snippet generate(metadata)
+  Gemfile.lock
+  coverage/*
```
Comment (swcloud) [inline src/main/resources/com/google/api/codegen/ruby/gitignore.snip]: Can you also IDE settings such as 

.idea
*.iml

Comment (landrito) [inline src/main/resources/com/google/api/codegen/ruby/gitignore.snip]: Done.

**Pull request merged**